### PR TITLE
fix: create and edit group/channel public switcher

### PR DESCRIPTION
**Pull request by MatveyNeyman**
Create - remember switcher position when going to next screen
Edit - disable switcher if public

### CODE REVIEW

File: src/components/CreateNewModal/CreateNewModal.js [hunk 1]
```diff
@@ -23,13 +23,25 @@ import ImageEdit from '../ImageEdit/ImageEdit';
...
   static defaultProps = {
     id: 'create_new_modal',
     isPublicGroupsEnabled: true,
     isMaxGroupSizeVisible: false,
   };
 
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
```
Comment (meafmira) [inline src/components/CreateNewModal/CreateNewModal.js]: Move to class property

File: src/components/CreateNewModal/CreateNewModal.js [hunk 2]
```diff
@@ -23,13 +23,25 @@ import ImageEdit from '../ImageEdit/ImageEdit';
 import styles from './CreateNewModal.css';
 import HotKeys from '../HotKeys/HotKeys';
 
-class CreateNewModal extends PureComponent<Props> {
+type State = {
```
Comment (meafmira) [inline src/components/CreateNewModal/CreateNewModal.js]: Name `CreateNewModalState`

File: src/components/EditGroupModal/EditGroupModalForm.js [hunk 3]
```diff
@@ -57,21 +58,17 @@ class EditGroupModalForm extends PureComponent<Props, State> {
   constructor(props: Props, context: Context) {
     super(props, context);
 
+    this.isPublic = props.shortname && Boolean(props.shortname.value);
```
Comment (meafmira) [inline src/components/EditGroupModal/EditGroupModalForm.js]: No. This is wrong. Don't copy props to class fields. Calculate it every time when use. Or move to derived state

File: src/components/CreateNewModal/CreateNewModal.js (hunk 1 above)
Comment (MatveyNeyman) [reply src/components/CreateNewModal/CreateNewModal.js]: fixed

File: src/components/CreateNewModal/CreateNewModal.js (hunk 2 above)
Comment (MatveyNeyman) [reply src/components/CreateNewModal/CreateNewModal.js]: fixed

File: src/components/EditGroupModal/EditGroupModalForm.js [hunk 4]
```diff
@@ -58,17 +59,20 @@ class EditGroupModalForm extends PureComponent<Props, State> {
   constructor(props: Props, context: Context) {
     super(props, context);
 
-    this.isPublic = props.shortname && Boolean(props.shortname.value);
+    const isPublic = props.shortname && Boolean(props.shortname.value);
+    const shortname = props.shortname.value || '';
 
     if (!props.avatar || typeof props.avatar === 'string') {
```
Comment (meafmira) [inline src/components/EditGroupModal/EditGroupModalForm.js]: I don't understand what are you doing here. Why not `if (typeof props.avatar === 'string')`?

File: src/components/EditGroupModal/EditGroupModalForm.js [hunk 5]
```diff
@@ -102,6 +106,15 @@ class EditGroupModalForm extends PureComponent<Props, State> {
...
     this.setState({ isPublic });
+    this.props.onPublicToggle(isPublic);
+  };
+
+  handleShortnameChange = (
+    shortname: string,
+    event: SyntheticInputEvent<>,
+  ): void => {
+    this.setState({ shortname });
+    this.props.onChange(shortname, event);
```
Comment (meafmira) [inline src/components/EditGroupModal/EditGroupModalForm.js]: Why you need this? And why it was named `onChange` not `onShortnameChange`?

File: src/components/EditGroupModal/EditGroupModalForm.js (hunk 4 above)
Comment (MatveyNeyman) [reply src/components/EditGroupModal/EditGroupModalForm.js]: @meafmira I didn't change this line as you may see. As a kind suggestion please consider refactor this file as a separate task.

**Pull request merged**